src/components/select/__internal__/select-list/select-list.test.tsx

```diff
@@ -0,0 +1,1020 @@
+        <Option id="green" value="green" text="green" />
+        <Option id="blue" value="blue" text="blue" />
+      </SelectListWithInput>
+    );
+
+    expect(screen.getByRole("option", { name: /red/i })).toHaveAttribute(
+      "aria-selected",
+      "false"
+    );
+    expect(screen.getByRole("option", { name: /green/i })).toHaveAttribute(
+      "aria-selected",
+      "false"
+    );
+    expect(screen.getByRole("option", { name: /blue/i })).toHaveAttribute(
+      "aria-selected",
+      "false"
+    );
+  });
+});
+
+test("no error thrown when no options are passed but a highlightedValue is provided", () => {
+  expect(() =>
+    render(<SelectListWithInput highlightedValue="red" />)
+  ).not.toThrow();
+});
+
+test("hides the selected option if it is the only list item and the list is loading.", () => {
+  render(
+    <SelectListWithInput highlightedValue="green" isLoading>
+      <Option id="green" value="green" text="green" />
+    </SelectListWithInput>
+  );
+
+  const hiddenOption = screen.getByRole("option", {
+    hidden: true,
+  });
+  expect(hiddenOption).toHaveTextContent("green");
+});
+
+describe("keyboard navigation", () => {
+  it("calls onSelect to confirm selection when option is selected by pressing Enter key", async () => {
+    const onSelect = jest.fn();
+    const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+
+    render(
+      <SelectListWithInput onSelect={onSelect} highlightedValue="red">
+        <Option id="red" value="red" text="red" />
+        <Option id="green" value="green" text="green" />
+        <Option id="blue" value="blue" text="blue" />
+      </SelectListWithInput>
+    );
+
+    await user.keyboard("{Enter}");
+
+    expect(onSelect).toHaveBeenCalledWith(
+      expect.objectContaining({
+        value: "red",
+        selectionConfirmed: true,
+        selectionType: "enterKey",
+      })
+    );
+  });
+
+  it("calls onSelect to confirm selection when option row is selected by pressing Enter key", async () => {
+    const onSelect = jest.fn();
+    const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+
+    render(
+      <SelectListWithInput
+        onSelect={onSelect}
+        highlightedValue="dark-red"
+        multiColumn
+      >
+        <OptionRow id="dark-red" value="dark-red" text="dark red">
+          <td>dark</td>
+          <td>red</td>
+        </OptionRow>
+        <OptionRow id="light-blue" value="light-blue" text="light blue">
+          <td>light</td>
+          <td>blue</td>
+        </OptionRow>
+      </SelectListWithInput>
+    );
+
+    await user.keyboard("{Enter}");
+
+    expect(onSelect).toHaveBeenCalledWith(
+      expect.objectContaining({
+        value: "dark-red",
+        selectionConfirmed: true,
+        selectionType: "enterKey",
+      })
+    );
+  });
+
+  it("if the highlighted option is somehow disabled, pressing Enter key does not call onSelect", async () => {
```

Comment:
**nitpick(non-blocking):** "somehow" makes it seem like we don't know why the `Option` is disabled but we're passing it the `disabled` prop.

```suggestion
  it("if the highlighted option is disabled, pressing Enter key does not call onSelect", async () => {
```